### PR TITLE
Let events be able to animate geodesy and seismology symbols

### PR DESCRIPTION
**Pull request by PaulWessel**
**Description of proposed changes**

This PR adds a new option **-Z**_command_ to **events** that is used to pass the core command needed to one of **meca**, **coupe**, or **velo**.  Here, _command_ will be an abbreviated core option set that specifies the desired symbol that the selected supplemental module can draw.  An example of a **-Z** option may be **-Z**"meca  -Sa5c+f0" given in quotes.  Other **meca** options such as **-C**, **-G**, **-W** are common to **events** and thus given on the **events** command line and not part of the **meca** command string.  As a minimal example, here is a working main.sh script:

```
gmt begin
	gmt basemap -JM6i -R0/6/0/6 -B -X0.5i -Y0.5i
	gmt events data.txt -Z"psmeca -Sa5c+f0" -Gred -W1p -T${MOVIE_COL0} -Es+r1+p1+d1+f1 -Mi1+c-0.5 -Ms2+c1 -Mt+c0 --TIME_UNIT=d
gmt end show
```

driven by the **movie** command 

`gmt movie main.sh -C7ix7ix100 -Nmv -Fmp4 -Ml -Lc0+o2c -Ttime.txt`

where _time.txt_ should first be created via

`gmt math -T2020-01-31T/2020-02-04T/0.2 --TIME_UNIT=h -o0 T = time.txt`

and the data.txt file is simply

`3 3 0  4.319 37.28 145.8  6 2020-02-01T14:00:000 0 100% double-couple [-Sa]`

Since @Esteban82 was the one to request this feature I hope he has time to give this branch some testing since there are likely some minor things we need to address.  But as you can see at the end of #4866 it is off to a good start.

### CODE REVIEW

**Comment by Esteban82** [comment]
Thanks @Paul. I try this command and works fine ([movie link](https://www.dropbox.com/s/66ybkmo9kjjqd1s/EJ24.2_Anim_Mapa_ARG_Sismicidad_Mecanismos.mp4?dl=0)). 

BTW, I have to add a column with time in the [globalcmt data](https://www.globalcmt.org/CMTsearch.html). Maybe you could contact them to ask them for a "GMT psmeca input with time" output format. 

I use this command:	

`gmt events "Datos_Mec.txt" -Z"psmeca -Sd0.5c+f0" -Gred -W1p -T\${MOVIE_COL0} -Es+r2+p1+d6+f1 -Mi1+c-0.5 -Ms1.3+c1 -Mt+c90 --TIME_UNIT=o	`


Master frame:  EJ24 2_Anim_Mapa_ARG_Sismicidad_Mecanismos

**Comment by PaulWessel** [comment]
> Maybe you could contact them to ask them for a "GMT psmeca input with time" output format.

Thanks. I sent Ekstrom and Nettles a request.  They are along-time GMT users and I know them so hopefully it is not too much to ask.

**Comment by PaulWessel** [comment]
I should at least make two more test animations exercising **coupe** and **velo** before I remove the WIP. I also need to understand if the sudden change in color during the code is missing a ramp or not.  Hopefully today.  And perhaps we should see if @Esteban82 test movie could be converted into a new animation 13 or 14 to highlight this capability more broadly.

**Comment by Esteban82** [comment]
I would be glad to contribute with a animation 13/14.  I just would need to find a better example (maybe an earthquake swarm) and with real data (in the previous example the time data is not correct).

**Comment by PaulWessel** [comment]
Great, take your time.  I suggest you look at the others (you already have) and construct something using the HD canvas and make it long enough and of high-enough resolution that it becomes another YouTube posting.  Ideally, the script would grab the data from the internet via an URL query like we do for one of the other examples.
Unrelated to this, consider using one of the scientific color maps since we are getting criticized for doing too much rainbow, jet, etc...

**Comment by PaulWessel** [comment]
Ekstrom says they will have a look at this once the semester dies down.  I expect to have to ping him in June...

**Comment by PaulWessel** [comment]
Made various minor fixes to ensure coupe can be animated as well:

https://user-images.githubusercontent.com/26473567/110582231-faf84800-810f-11eb-939a-9b9171e97bfb.mp4

**Comment by PaulWessel** [comment]
The abrupt change in intensity is due to me not setting an event duration with -L.  My final scripts will do the right thing.

**Comment by PaulWessel** [comment]
I have finished the implementation for velo as well.  Here is a simple movie showing the basics:

https://user-images.githubusercontent.com/26473567/110727138-c42d3b00-81be-11eb-95ff-3e0cae0a9742.mp4

Note: It may not make sense to scale the lengths here since they reflect measurements, but that is up to the user via -**Ms**.  I am just showing that the scaling, intensity, and transparency features work.

**Comment by PaulWessel** [comment]
I have now finalized this PR and turned my three test-scripts in to actual tests in seis and geodesy.  While they are animation scripts, they only generate the master frame PS file, but during debug they can be used for movies.  Here are the plots:

**geodesy_08.sh:**

geodesy_08

**seis_11.sh:**

seis_11

**seis_12.sh:**

seis_12

The concerns raised in #4941 can be dealt with in later PRs.  Asking for formal reviews from @joa-quim, @meghanrjones and @seisman.

**Review by meghanrjones — COMMENTED** [review]
@PaulWessel, these test results need to be updated to account for themes:
	315 - test/geodesy/geodesy_08.sh (Failed)
	980 - test/seis/seis_11.sh (Failed)
	981 - test/seis/seis_12.sh (Failed)

**Comment by PaulWessel** [comment]
OK, did not look yet.  Are we just talking about updating the PS or do I need to compare and make changes to the scripts?

**Comment by meghanrjones** [comment]
> OK, did not look yet. Are we just talking about updating the PS or do I need to compare and make changes to the scripts?

If these were included in the docs, it would be necessary to add ``gmt set GMT_THEME cookbook``. But since these are tests no changes are needed to the scripts.

**Comment by PaulWessel** [comment]
I will run and have a look first before I update.  Once finished with another branch check...  I head that 16-core M2 chip from the future.

File: doc/rst/source/events.rst [hunk 1]
```diff
@@ -35,6 +35,7 @@ Synopsis
 [ |-W|\ *pen* ]
 [ |SYN_OPT-X| ]
 [ |SYN_OPT-Y| ]
+[ |-Z|\ "custom command" ]
```
Comment (meghanrjones) [inline doc/rst/source/events.rst:38]: The "custom command" seems stylistically different to me from other flags. Do you think it's more clear than describing the arguments to **-Z** as *command*. 
```suggestion
[ |-Z|\ *command* ]
```

File: doc/rst/source/events_common.rst_ [hunk 2]
```diff
@@ -204,6 +204,22 @@ Optional Arguments
 
 .. include:: explain_-XY.rst_
 
+.. _-Z:
+
+**-Z**\ "custom command"
+    Place symbols generated by either :doc:`/supplements/seis/coupe`,
+    :doc:`/supplements/seis/meca`, or :doc:`/supplements/geodesy/velo`.
+    The quoted custom command must start with one of these module names, then
```
Comment (meghanrjones) [inline doc/rst/source/events_common.rst_:212]:
```suggestion
    The quoted *command* must start with one of these module names, then
```

File: doc/rst/source/events_common.rst_ [hunk 3]
```diff
@@ -204,6 +204,22 @@ Optional Arguments
...
+    supply all the module-specific options that the selected module requires to
+    normally define a plot, excluding any of **-C**, **-G**, **-J**, **-N**, **-R**, and
+    **-W** (those options may be required by **events** though).  Also, you may
+    not use the **-I** or **-t** options since these will be set automatically
+    as part of the variations imposed by **-Mi** and **-Mt**. **Note**: If
+    you are running a simple classic command then you must use classic module names.
+    As an example, the custom command to plot a beachball may be **-Z**\ "meca -Sa5c+f0",
+    while displaying a crossection of one may require the more elaborate command
+    **-Z**\ "coupe -Q -L -Sc3c -Ab128/11/120/250/90/400/0/100+f -Fa0.1i/cc".
```
Comment (meghanrjones) [inline doc/rst/source/events_common.rst_:221]:
```suggestion
    as part of the variations imposed by **-Mi** and **-Mt**.  As an example, the
    custom command to plot a beachball may be **-Z**\ "meca -Sa5c+f0", while
    displaying a crossection of one may require the more elaborate command
    **-Z**\ "coupe -Q -L -Sc3c -Ab128/11/120/250/90/400/0/100+f -Fa0.1i/cc".
    **Note**: If you are running a simple classic command then you must use
    classic module names.
```
It makes more sense to me to but the classic command note after the modern mode example.

File: doc/rst/source/psevents.rst [hunk 4]
```diff
@@ -37,6 +37,7 @@ Synopsis
 [ |-W|\ *pen* ]
 [ |SYN_OPT-X| ]
 [ |SYN_OPT-Y| ]
+[ |-Z|\ "custom command" ]
```
Comment (meghanrjones) [inline doc/rst/source/psevents.rst:40]:
```suggestion
[ |-Z|\ *command* ]
```

File: test/geodesy/geodesy_08.sh [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+-124.6	40	2020-02-04T		FADE
+-124.6	40	2020-02-05T		CODA
+EOF
+gmt math -T2020-01-31T/2020-02-06T/2 --TIME_UNIT=h -o0 T = time.txt
+cat << EOF > main.sh
+gmt begin
+	gmt basemap -R-124.6/-123.3/40/41 -JM5i -B -BWSne -Y0.5i -X0.75i --MAP_FRAME_TYPE=plain
+	gmt events labels.txt -L1 -Et -T\${MOVIE_COL0} -F+f12p,Helvetica-Bold,red+jBL -Dj0.1c --TIME_UNIT=d
+	gmt events -Z"velo -A18p+e+a40+n6k+h0.5 -Se0.50/0/0p" -Ct.cpt -W1.5 -N data.txt -T\${MOVIE_COL0} -Es+r1+p1+d1+f1 -Mi1+c-0.5 -Ms2+c0.5 -Mt+c0 -L3 --TIME_UNIT=d
+gmt end show
```
Comment (meghanrjones) [inline test/geodesy/geodesy_08.sh:43]:
```suggestion
gmt end
```

File: test/seis/seis_11.sh [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+0	0	2020-02-04T		FADE
+0	0	2020-02-05T		CODA
+EOF
+gmt math -T2020-01-31T/2020-02-06T/2 --TIME_UNIT=h -o0 T = time.txt
+cat << EOF > main.sh
+gmt begin
+	gmt basemap -JM5i -R0/6/0/6 -B -X0.5i -Y0.5i
+	gmt events data.txt -Z"psmeca -Sa4c+f0" -Ct.cpt -W1p -T\${MOVIE_COL0} -Es+r1+p1+d1+f1 -Mi1+c-0.5 -Ms2+c0.5 -Mt+c0 -L3 --TIME_UNIT=d
+	gmt events labels.txt -L1 -Et -T\${MOVIE_COL0} -F+f12p,Helvetica-Bold,red+jBL -Dj0.1c --TIME_UNIT=d
+gmt end show
```
Comment (meghanrjones) [inline test/seis/seis_11.sh:24]:
```suggestion
gmt end
```

File: test/seis/seis_12.sh [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+0	100	2020-02-04T		FADE
+0	100	2020-02-05T		CODA
+EOF
+gmt math -T2020-01-31T/2020-02-06T/2 --TIME_UNIT=h -o0 T = time.txt
+cat << EOF > main.sh
+gmt begin
+	gmt basemap -R0/250/0/100 -Jx0.02i/-0.02i -B -Y0.5i -X0.5i
+	gmt events labels.txt -L1 -Et -T\${MOVIE_COL0} -F+f12p,Helvetica-Bold,red+jBL -Dj0.1c --TIME_UNIT=d
+	gmt events -Z"coupe -Q -L -Sc3c -Ab128/11/120/250/90/400/0/100+f -Fa0.1i/cc" -Ct.cpt -N data.txt -T\${MOVIE_COL0} -Es+r1+p1+d1+f1 -Mi1+c-0.5 -Ms2+c0.5 -Mt+c0 -L3 --TIME_UNIT=d
+gmt end show
```
Comment (meghanrjones) [inline test/seis/seis_12.sh:26]:
```suggestion
gmt end
```

**Review by meghanrjones — APPROVED** [review]
Looks good to me.

**Comment by Esteban82** [comment]
> And perhaps we should see if @Esteban82 test movie could be converted into a new animation 13 or 14 to highlight this capability more broadly.

Should I open a pull request ("WIP Animation 14")?

**Comment by PaulWessel** [comment]
Yes please.  We are close to finishing up a new anim13 so call in anim14 and do it as a WIP for now.

**Comment by Esteban82** [comment]
@PaulWessel it would be possible that I am not allow to do that? It seems to me that I do not have the permissions to create it.

**Comment by PaulWessel** [comment]
Hi @meghanrjones and @seisman. I would like to lower the barrier for @Esteban82 to contribute.  He should be able to do this without too much hassle.  He has proven very helpful and trustworthy I think

**Comment by seisman** [comment]
I just give the [gmt-contributors team](https://github.com/orgs/GenericMappingTools/teams/gmt-contributors) write permission to the gmt repository. @Esteban82 is in this team and should be able to create branches now. We will do more tweaks to teams and team permissions after more discussions in https://github.com/GenericMappingTools/gmt/discussions/4732.

@Esteban82 Please let me know if it works for you.

**Pull request merged**